src/Controller/CRUDController.php

```diff
@@ -1612,4 +1597,17 @@ private function setFormTheme(FormView $formView, array $theme = null): void
 
         $twig->getRuntime(FormRenderer::class)->setTheme($formView, $theme);
     }
+
+    private function handleXmlHttpRequestErrorResponse(FormInterface $form): Response
```

Comment:
Does it return a Response or a JsonResponse?

Reply:
Autogenerated by IDE because `renderJson` has `@return Response` phpdoc. I'll add `JsonResponse` return-type.